### DISCUSSION TITLE
App Router Subpath Folder

### DISCUSSION THREAD

**Discussion by kudla** [vercel/next.js#71928] [Help] [answered]
### Summary

Hi
Trying to achieve routing hierarchy e.g. `/section/sub-section` I'm mapping this to 
* `/app/section/pages.tsx`
* `/app/section/sub-section/pages.tsx`

Still building exported application at the `out` folder it renders the endpoint paths as 
* `/section.html`
* `/section/nested.html`

So is it possible to make app routing to compose them as  indexed folders instead e.g.
* `/section/index.html`
* /section/nested/index.html`

As I understood using `/index.tsx` is kind of legacy `pages` routing way? Still long time ago it worked exactly as `folder/index.html` way.

And this is exactly the pretty thing for serving the static app from `s3` bucket  e.g.

So is there a way to achieve the same with `app` routing. Or should I really downgrade for `page` routing (or even for the legacy `next` versions).

Thank you

**Comment by aniruddhaadak80** [comment, marked as answer]
Hi, @kudla 

---

To achieve the routing you want (e.g., `/section` as `/section/index.html`), here are a few suggestions:

1. **Folder Structure**: Make sure your folder structure is:
   - `/app/section/page.tsx`
   - `/app/section/sub-section/page.tsx`

   This should help create the routes without the `.html` suffix.

2. **Custom Rewrites**: If you still want the `.html` extension, consider adding rewrite rules in your `next.config.js`:

   ```javascript
   module.exports = {
     async rewrites() {
       return [
         { source: '/section', destination: '/section/index.html' },
         { source: '/section/sub-section', destination: '/section/nested/index.html' },
       ];
     },
   };
   ```

3. **Static Export**: Use `next export` to ensure the pages are properly exported for static hosting.

4. **S3 Setup**: If serving from S3, set the index document to `index.html` so it serves automatically when accessing a folder.

If you need more help, feel free to share additional details!

Thanks!

**Reply by kudla** [reply]
Thank you @aniruddhaadak80

**Reply by aniruddhaadak80** [reply]
I am glad that I could help you @kudla . 🎉
